include/bm/bm_sim/queueing.h

```diff
@@ -578,7 +578,7 @@ class QueueingLogicPriRL {
       } else {
         auto now = clock::now();
         auto next = clock::time_point::max();
-        for (pri = 0; pri < nb_priorities; pri++) {
+        for (pri = nb_priorities - 1; 0 <= pri; pri--) {
```

Comment:
If `size_t` is an unsigned type, then note that `0 <= pri` is _always_ true, and this becomes an infinite loop unless something prevents that in the loop body.  It might be best to change the declaration of `pri` to `int pri;` ?

Reply:
According to a warning during compilation in one of the CI tests, `0 <= pri` is always true, so `size_t` is definitely an unsigned type and that declaration of `pri` should be changed to a signed type.

Reply:
this looks cool:
```cpp
for (pri = nb_priorities; pri --> 0 ;)
```
source: https://stackoverflow.com/a/3626468

it will avoid writing a cast...